packages/base/src/panelview/objectproperties.tsx

```diff
@@ -256,98 +234,66 @@ class ObjectPropertiesReact extends React.Component<IProps, IStates> {
     if (selectedObj.type) {
       schema = this._formSchema.get(selectedObj.type);
 
-      // Generate dropdown for layer source entry
-      if (
-        schema &&
-        schema.properties.source &&
-        selectedObjectData &&
-        selectedObjectData.source
-      ) {
-        const sourceNames: string[] = [];
-        for (const sourceId of Object.keys(
-          this.props.cpModel.jGISModel?.getSources() || {}
-        )) {
-          const source = this.props.cpModel.jGISModel?.getSource(sourceId);
-          if (source) {
-            sourceNames.push(source.name);
-          }
-        }
+      if (selectedObjectData && selectedObjectData.source) {
         selectedObjectSourceId = selectedObjectData.source;
-        selectedObjectData.source = this.props.cpModel.jGISModel?.getSource(
-          selectedObjectData.source
-        )?.name;
-        schema.properties.source.enum = sourceNames;
       }
     }
 
-    let sourceSchema: IDict<any> | undefined;
-    let selectedObjectSourceData: IDict<any> | undefined;
-    if (selectedObjSource) {
-      sourceSchema = this._formSchema.get(selectedObjSource.type);
-      selectedObjectSourceData = selectedObjSource.parameters;
+    if (!selectedObjectData) {
+      return <div></div>;
     }
 
-    return {
-      ...old,
-      selectedObjectData,
-      selectedObject,
-      selectedObjectType: selectedObjSource === undefined ? 'source' : 'layer',
-      selectedObjectSource: selectedObjectSourceId,
-      schema,
-      selectedObjectSourceData,
-      sourceSchema
-    };
-  }
+    // We selected a source and only show a form for it
+    if (!selectedObjSource) {
+      return (
+        <div>
+          <h3>Layer Properties</h3>
+          <LayerPropertiesForm
```

Comment:
I'm confused between the comment , the condition and the component.
I agree with the comment though, we should be able to select only a source.

Reply:
Now that you point it out, I'm confused too. I'll look into it/